tests/zim/test_zim_creator.py

```diff
@@ -539,6 +541,18 @@ def test_check_metadata(tmp_path):
     with pytest.raises(ValueError, match="LongDescription is too long."):
         Creator(tmp_path, "").config_dev_metadata(LongDescription="T" * 5000).start()
 
+def test_start_logs_metadata_debug_enabled(tmp_path, mocker):
+    # logger is not resolving?
+    #with mocker.patch.object(logger, 'isEnabledFor', new=MagicMock(return_value=True)) as faux_logger_debug:
+     #   with mocker.spy(faux_logger_debug, 'debug') as faux_logger:
```

Comment:
I haven't done a lot with python mocks yet. Is there a better way to do these two patches?

Reply:
The kind of stuff below is probably sufficient:

```
from unittest.mock import patch

@patch("zimscraperlib.getLogger")
@patch("zimscraperlib.zim.creator.logger")
def test_start_logs_metadata_debug_enabled(mocked_get_logger, _, tmp_path):
    mocked_get_logger.isEnabledFor.return_value = True
    Creator(tmp_path, "").start()
    mocked_get_logger.info.assert_called_with("This is an info message")


@patch("zimscraperlib.getLogger")
@patch("zimscraperlib.zim.creator.logger")
def test_start_logs_metadata_debug_disabled(mocked_get_logger, _, tmp_path):
    mocked_get_logger.isEnabledFor.return_value = False
    Creator(tmp_path, "").start()
    mocked_get_logger.info.assert_not_called()
```

You might even want to use more precise mocking by replacing `mocked_get_logger.isEnabledFor.return_value = True` with `mocked_get_logger.isEnabledFor.side_effect = lambda level: level == logging.DEBUG`